python/lsst/obs/base/makeRawVisitInfoViaObsInfo.py

```diff
@@ -0,0 +1,172 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import astropy.units
+from astropy.utils import iers
+
+# This is an unofficial ERFA interface
+import astropy._erfa as erfa
+
+from lsst.log import Log
+from lsst.daf.base import DateTime
+from lsst.geom import degrees, radians
+from lsst.afw.image import VisitInfo, RotType
+from lsst.afw.coord import Observatory, Weather
+from lsst.geom import SpherePoint
+from lsst.obs.metadata import ObservationInfo
+
+__all__ = ["MakeRawVisitInfoViaObsInfo"]
+
+
+class MakeRawVisitInfoViaObsInfo(object):
+    """Base class functor to make a VisitInfo from the FITS header of a
+    raw image using `~lsst.obs.metadata.ObservationInfo` translators.
+
+    Subclasses can be used if a specific
+    `~lsst.obs.metadata.MetadataTranslator` translator should be used.
+
+    The design philosophy is to make a best effort and log warnings of
+    problems, rather than raising exceptions, in order to extract as much
+    VisitInfo information as possible from a messy FITS header without the
+    user needing to add a lot of error handling.
+
+    Parameters
+    ----------
+    log : `lsst.log.Log` or None
+        Logger to use for messages.
+        (None to use ``Log.getLogger("MakeRawVisitInfoViaObsInfo")``).
+    """
+
+    metadataTranslator = None
+    """Header translator to use to construct VisitInfo, defaulting to
+    automatic determination."""
+
+    def __init__(self, log=None):
+        if log is None:
+            log = Log.getLogger("MakeRawVisitInfoViaObsInfo")
+        self.log = log
+
+    def __call__(self, md, exposureId=None):
+        """Construct a VisitInfo and strip associated data from the metadata.
+
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertyList` or `lsst.daf.base.PropertySet`
+            Metadata to pull from.
+            Items that are used are stripped from the metadata.
+        exposureId : `int`, optional
+            Ignored.  Here for compatibility with `MakeRawVisitInfo`.
+
+        Returns
+        -------
+        visitInfo : `lsst.afw.image.VisitInfo`
+            `~lsst.afw.image.VisitInfo` derived from the header using
+            a `~lsst.obs.metadata.MetadataTranslator`.
+        """
+        argDict = dict()
+
+        obsInfo = ObservationInfo(md, translator_class=self.metadataTranslator)
+        print(obsInfo.__dict__)
+
+        # Strip all the cards out that were used
+        for c in obsInfo.cards_used:
+            del md[c]
```

Comment:
Could/should this operation be an `ObservationInfo` method?

Reply:
It actually is in `ObservationInfo`, except the method there deliberately returns a copy of the supplied metadata stripped of headers rather than stripping in place.  It seemed a bit naughty to modify in place.